src/test/java/com/sharewalk/controller/WalkControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package com.sharewalk.controller;
+
+import com.sharewalk.dao.WalkDAO;
+import com.sharewalk.model.Walk;
+import com.sharewalk.service.WalkService;
+import junitparams.JUnitParamsRunner;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnitParamsRunner.class)
+public class WalkControllerTest {
+
+    @InjectMocks
+    private WalkController instance;
+
+    @Mock
+    private WalkDAO walkDAO;
+
+    @Mock
+    private WalkService walkService;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
```

Comment:
You don't need initMocks method if you use @InjectMocks annotation. Without @Inject mock you would have to do initMocks and to put mocks into walkController